packages/xo-server-backup-reports/src/index.js

```diff
@@ -146,72 +221,168 @@ class BackupReportsXoPlugin {
   }
 
   unload() {
     this._xo.removeListener('job:terminated', this._report)
   }
 
-  _wrapper(status, job, schedule, runJobId) {
-    if (job.type === 'metadataBackup') {
-      return
-    }
+  async _report(runJobId, { type, status } = {}, force) {
+    const xo = this._xo
+    try {
+      if (type === 'call') {
+        return this._legacyVmHandler(status)
+      }
 
-    return new Promise(resolve =>
-      resolve(
-        job.type === 'backup'
-          ? this._backupNgListener(status, job, schedule, runJobId)
-          : this._listener(status, job, schedule, runJobId)
-      )
-    ).catch(logError)
+      const log = await xo.getBackupNgLogs(runJobId)
+      if (log === undefined) {
+        throw new Error(`no log found with runId=${JSON.stringify(runJobId)}`)
+      }
+
+      const reportWhen = log.data.reportWhen
+      if (
+        !force &&
+        (reportWhen === 'never' ||
+          (reportWhen === 'failure' && log.status === 'success'))
+      ) {
+        return
+      }
+
+      const [job, schedule] = await Promise.all([
+        await xo.getJob(log.jobId),
+        await xo.getSchedule(log.scheduleId).catch(error => {
+          logger.warn(error)
+        }),
+      ])
+
+      if (job.type !== 'backup' && job.type !== 'metadataBackup') {
+        throw new Error('Unknown backup job type')
+      }
+
+      return job.type === 'backup'
+        ? this._ngVmHandler(log, job, schedule, force)
+        : this._metadataHandler(log, job, schedule, force)
+    } catch (error) {
+      logger.warn(error)
+    }
   }
 
-  async _backupNgListener(_1, _2, schedule, runJobId) {
+  async _metadataHandler(log, { name: jobName }, schedule, force) {
     const xo = this._xo
-    const log = await xo.getBackupNgLogs(runJobId)
-    if (log === undefined) {
-      throw new Error(`no log found with runId=${JSON.stringify(runJobId)}`)
-    }
 
-    const { reportWhen, mode } = log.data || {}
-    if (
-      reportWhen === 'never' ||
-      (log.status === 'success' && reportWhen === 'failure')
-    ) {
-      return
+    const formatDate = createDateFormatter(schedule?.timezone)
+
+    const tasksByStatus = groupBy(log.tasks, 'status')
+    const n = log.tasks?.length ?? 0
+    const nSuccesses = tasksByStatus.success?.length ?? 0
+
+    if (!force && log.data.reportWhen === 'failure') {
+      delete tasksByStatus.success
     }
 
-    if (schedule === undefined) {
-      schedule = await xo.getSchedule(log.scheduleId)
+    // header
+    const markdown = [
+      `##  Global status: ${log.status}`,
+      '',
+      `- **Job ID**: ${log.jobId}`,
+      `- **Job name**: ${jobName}`,
+      `- **Run ID**: ${log.id}`,
+      ...getTemporalDataMarkdown(log.end, log.start, formatDate),
+      n !== 0 && `- **Successes**: ${nSuccesses} / ${n}`,
+      ...getWarningsMarkdown(log.warnings),
+      getErrorMarkdown(log),
+    ]
+
+    const nagiosText = []
+
+    // body
+    for (const status in tasksByStatus) {
```

Comment:
In which order are they processed?

Reply:
They are processed randomly.

Reply:
This is not good enough, we should control in which order they are displayed.

Reply:
They are grouped by status. Should we display them on this order failures -> interrupted -> skipped -> success ?

Reply:
It seems reasonable.

Random order is never the good answer, we should decide which order to use and be consistent.